src/app/dynamic-forms/fields-wrapper/fields-wrapper.component.ts

```diff
@@ -1,29 +1,61 @@
-import {Component, EventEmitter, Input, OnInit, Output, ViewEncapsulation} from '@angular/core';
-import {AppStateService} from "../../services/app-state.service";
+import { Component, EventEmitter, Input, OnChanges, OnInit, Output, ViewEncapsulation } from '@angular/core';
+import { AppStateService } from '../../services/app-state.service';
 
 @Component({
-  selector: 'app-fields-wrapper',
-  templateUrl: './fields-wrapper.component.html',
-  styleUrls: ['./fields-wrapper.component.scss'],
+  selector     : 'app-fields-wrapper',
+  templateUrl  : './fields-wrapper.component.html',
+  styleUrls    : ['./fields-wrapper.component.scss'],
   encapsulation: ViewEncapsulation.None
 })
-export class FieldsWrapperComponent implements OnInit {
+export class FieldsWrapperComponent implements OnInit, OnChanges {
   @Input() form;
   @Input() rowId;
   @Input() formSections;
   @Input() isChild: boolean;
   @Input() fieldsWithErrors;
   @Input() hasChildTableAccess: boolean;
+  @Input() formFieldMetaDatas: any = {};
   @Output() eventOccurred = new EventEmitter<any>();
   @Output() childFormOpened = new EventEmitter<any>();
   @Output() deleteDialogOpened = new EventEmitter<any>();
+  public showSpinner = false;
+
   constructor(private appStateService: AppStateService) { } // AppState service is outside of Dynamic forms
 
   ngOnInit(): void {
   }
 
   expansionClicked(section) {
+    if(this.form)
+    {
+      if(this.form.sections[1]){
+        if(this.form.sections[1].fields.length == 0)
+        {
+          this.showSpinner = true;
+        }
+        else
+        {
+          this.showSpinner = false;
+        }
+      }
+    }
     this.appStateService.sectionClicked(section.label);
   }
 
+  ngOnChanges(): void {
```

Comment:
What is the logic in this function supposed to do?

Reply:
We added new spinner for non-expanded sections,
For set enableSaveBtn = true (to show new spinner which added for non expanded sections),
We added this.

Reply:
I'm just not sure if this is the right function to put this in, ngOnChanges() gets fired very often, not just when you expand a section. What specific change do you need in this component that will trigger this code?

Reply:
removed ngOnChanges() method